pkg/webhook/external/generic_list_options.go

```diff
@@ -0,0 +1,151 @@
+package external
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/pkg/expressions"
+	"github.com/akuity/kargo/pkg/expressions/function"
+)
+
+// buildListOptionsForTarget builds a list of client.ListOption based on the
+// provided GenericWebhookTarget's selectors. The returned ListOptions can be
+// used to list Kubernetes resources that match the target's selection criteria.
+func buildListOptionsForTarget(
+	project string,
+	t kargoapi.GenericWebhookTarget,
+	env map[string]any,
+) ([]client.ListOption, error) {
+	listOpts := []client.ListOption{client.InNamespace(project)}
+	if len(t.IndexSelector.MatchExpressions) > 0 {
```

Comment:
You can skip this conditional. The function you're calling appears to be safe even when `t.IndexSelector.MatchExpressions` is empty. The result of that call is safe to use with `append()`, even if it is empty.

Generally speaking, it's more idiomatic in Go to _not_ assert `len(x) > 0` before iterating over `x`. `range` is a de facto condition.